ArduPlane/mode_auto.cpp

```diff
@@ -159,3 +159,25 @@ bool ModeAuto::_pre_arm_checks(size_t buflen, char *buffer) const
     // Note that this bypasses the base class checks
     return true;
 }
+
+bool ModeAuto::is_landing() const
+{
+    switch(plane.mission.get_current_nav_cmd().id) {
+    case MAV_CMD_NAV_LAND:
+    case MAV_CMD_NAV_VTOL_LAND:
+    case MAV_CMD_DO_VTOL_TRANSITION:
+        return true;
+    }
+    return plane.mission.get_in_landing_sequence_flag();
+}
+
+bool ModeAuto::is_taking_off() const
+{
+    switch(plane.mission.get_current_nav_cmd().id) {
+    case MAV_CMD_NAV_TAKEOFF:
+    case MAV_CMD_NAV_VTOL_TAKEOFF:
+    case MAV_CMD_DO_VTOL_TRANSITION:
```

Comment:
This should not be here.

Reply:
"this" meaning MAV_CMD_DO_VTOL_TRANSITION or the whole switch?

Reply:
Sorry, Just `MAV_CMD_DO_VTOL_TRANSITION `.

Reply:
removed the whole switch